crates/openvino/src/property.rs

```diff
@@ -0,0 +1,135 @@
+    /// If optimize_size is selected, smaller cache files will be created.
+    /// If optimize_speed is selected, loading time will decrease but the cache file size will increase.
+    CacheMode,
+    /// The number of executor logical partitions.
+    NumStreams,
+    /// CPU affinity per thread.
+    Affinity,
+    /// The maximum number of threads that can be used for inference tasks.
+    InferenceNumThreads,
+    /// High-level OpenVINO hint for using CPU pinning to bind CPU threads to processors during inference.
+    HintEnableCpuPinning,
+    /// High-level OpenVINO hint for using hyper threading processors during CPU inference.
+    HintEnableHyperThreading,
+    /// High-level OpenVINO Performance Hints.
+    HintPerformanceMode,
+    /// High-level OpenVINO Hints for the type of CPU core used during inference.
+    HintSchedulingCoreType,
+    /// Hint for device to use specified precision for inference.
+    HintInferencePrecision,
+    /// Backs the Performance Hints by giving
+    /// additional information on how many inference requests the application will be
+    /// keeping in flight usually this value comes from the actual use-case  (e.g.
+    /// number of video-cameras, or other sources of inputs)
+    HintNumRequests,
+    /// Desirable log level.
+    LogLevel,
+    /// High-level OpenVINO model priority hint.
+    HintModelPriority,
+    /// Performance counters.
+    EnableProfiling,
+    /// Device Priorities config option,
+    /// with comma-separated devices listed in the desired priority.
+    DevicePriorities,
+    /// High-level OpenVINO Execution hint
+    /// unlike low-level properties that are individual (per-device), the hints are something that every device accepts
+    /// and turns into device-specific settings
+    /// Execution mode hint controls preferred optimization targets (performance or accuracy) for given model
+    ///
+    /// It can be set to be below value:
+    /// - `"PERFORMANCE"`: Optimize for max performance
+    /// - `"ACCURACY"`: Optimize for max accuracy
+    HintExecutionMode,
+    /// Whether to force terminate TBB when OV Core is destroyed.
+    ForceTbbTerminate,
+    /// Configure `mmap()` use for model read.
+    EnableMmap,
+    /// ?
+    AutoBatchTimeout,
+    /// Arbitrary string property key.
+    Other(Cow<'static, str>),
+}
+
+impl AsRef<str> for PropertyKey {
+    fn as_ref(&self) -> &str {
+        match self {
+            PropertyKey::SupportedProperties => "SUPPORTED_PROPERTIES",
+            PropertyKey::AvailableDevices => "AVAILABLE_DEVICES",
+            PropertyKey::OptimalNumberOfInferRequests => "OPTIMAL_NUMBER_OF_INFER_REQUESTS",
+            PropertyKey::RangeForAsyncInferRequests => "RANGE_FOR_ASYNC_INFER_REQUESTS",
+            PropertyKey::RangeForStreams => "RANGE_FOR_STREAMS",
+            PropertyKey::DeviceFullName => "FULL_DEVICE_NAME",
+            PropertyKey::DeviceCapabilities => "OPTIMIZATION_CAPABILITIES",
```

Comment:
```suggestion
            PropertyKey::DeviceCapabilities => "DEVICE_CAPABILITIES",
```

Typo?

Reply:
This is consistent with [property keys](https://github.com/openvinotoolkit/openvino/blob/34caeefd07800b59065345d651949efbe8ab6649/src/bindings/c/src/ov_property.cpp#L15C13-L15C48) where these naming discrepancies between `device`<->`optimization` , `model`<->`network` exist